src/xcode/ENA/ENA/Source/Services/Exposure Transaction/AppleFilesWriter.swift

```diff
@@ -11,10 +11,16 @@ struct WrittenPackages {
 
 	var urls: [URL]
 	func cleanUp() {
-		let fileManager = FileManager()
-		for url in urls {
-			try? fileManager.removeItem(at: url)
+		guard let directoryURL = urls.first?.deletingLastPathComponent() else {
+			return
 		}
+		
+		let fileManager = FileManager()
+		Log.info("Removing: \(directoryURL)", log: .localData)
+		
+		// Remove the whole directory, instead of removing each file and than forget to remove the directory
+		try? fileManager.removeItem(at: directoryURL)
```

Comment:
Could you use removeOldKeypackages here?

Reply:
I would not do this, `removeOldKeypackages` is only a fallback and `AppleFilesWriter` handles Keypackages one at a time.